astropy/coordinates/sky_coordinate.py

```diff
@@ -172,6 +173,15 @@ def __init__(self, *args, **kwargs):
             raise ValueError('Cannot create a SkyCoord without data')
 
     @property
+    def info(self):
+        """
+        Mixin column information (attributes)
+        """
+        if not hasattr(self, '_info'):
+            self._info = ColumnInfo()
+        return self._info
```

Comment:
@taldcroft - a first comment only yet, but I thought that rather than duplicate code over all classes, one could do something like:
```
info = property(get_column_info())
```
where `get_column_info` is from `utils` as well and takes a class as an argument, i.e., something like:
```
def get_column_info(cls=ColumnInfo):
    def column_info(self):
        if not hasattr(self, '_info'):
            self._info = cls(self)
         return self._info
    return column_info
```
For `Quantity`, one would then have
```
    info = property(get_column_info(QuantityColumnInfo))
```

But I need to think through this a bit more. In particular, I worry `info` is a sufficiently general name that for some out-of-astropy classes it is going to be used already.